query/query_test.go

```diff
@@ -0,0 +1,168 @@
+package query_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/gormtestsupport"
+	"github.com/fabric8-services/fabric8-wit/query"
+	"github.com/fabric8-services/fabric8-wit/resource"
+	tf "github.com/fabric8-services/fabric8-wit/test/testfixture"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestQueryRepository struct {
+	gormtestsupport.DBTestSuite
+}
+
+func TestRunQueryRepository(t *testing.T) {
+	suite.Run(t, &TestQueryRepository{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
+}
+
+func (s *TestQueryRepository) TestCreateQuery() {
+	t := s.T()
+	resource.Require(t, resource.Database)
+	repo := query.NewQueryRepository(s.DB)
+	t.Run("success", func(t *testing.T) {
+		title := "My WI for sprin #101"
+		qs := `{"hello": "worold"}`
+		// given
+		fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1))
+		q := query.Query{
+			Title:   title,
+			Fields:  qs,
+			SpaceID: fxt.Spaces[0].ID,
+			Creator: fxt.Identities[0].ID,
+		}
+		// when
+		err := repo.Create(context.Background(), &q)
+		require.Nil(t, err)
+		// then
+		if q.ID == uuid.Nil {
+			t.Errorf("Query was not created, ID nil")
+		}
+		if q.CreatedAt.After(time.Now()) {
+			t.Errorf("Query was not created, CreatedAt after Now()?")
+		}
+		assert.Equal(t, title, q.Title)
+		assert.Equal(t, qs, q.Fields)
+	})
+
+	t.Run("fail", func(t *testing.T) {
+		t.Run("empty title", func(t *testing.T) {
+			title := ""
+			qs := `{"hello": "worold"}`
+			// given
+			fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1))
+			q := query.Query{
+				Title:   title,
+				Fields:  qs,
+				SpaceID: fxt.Spaces[0].ID,
+			}
+			// when
+			err := repo.Create(context.Background(), &q)
+			// then
+			require.NotNil(t, err)
+		})
+		t.Run("invalid query json", func(t *testing.T) {
+			title := "My WI for sprin #101"
+			qs := "non-json query"
```

Comment:
I couldn't not find a place in the repo's creation function that checks for valid query JSON. Can you point me to the correct location, please?

Reply:
`fields` column is **jsonb** and hence it accepts only valid json. nothing to be done more for that IMO hence no code in the repo for that.

Reply:
I do think there is a need to check for valid query syntax. See my answer to this https://github.com/fabric8-services/fabric8-wit/pull/1824/commits/8936aa1ff1018e648fff1a0e66be3355a8eef093#r159368461